### PR TITLE
several small changes MERGEOK

### PR DESCRIPTION
**Pull request by olemagnusnorum**
added to readme, deleted deployment.xml, and changed QueryDocumentProcessor to catch terms and not substrings

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: incremental-search/search-suggestions/README.md [hunk 1]
```diff
@@ -1 +1,65 @@
...
+
+Uses streaming search to for document matching.
+
+## Qick start
+
+**Clone sample-apps and go to search-suggestions**
+
+<pre data-test="exec">
+$ git clone --depth 1 https://github.com/vespa-engine/sample-apps.git
+$ cd incremental-search/search-suggestions
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:12]:
```suggestion
$ cd sample-apps/incremental-search/search-suggestions
```

File: incremental-search/search-suggestions/README.md [hunk 2]
```diff
@@ -1 +1,65 @@
...
+</pre>
+
+**Check if application is ready**
+<pre data-test="exec">
+$ curl -s --head http://localhost:8080/ApplicationStatus
+</pre>
+
+**Feed the example documents to the application**
+<pre data-test="exec">
+$ curl -L -o vespa-http-client-jar-with-dependencies.jar \ https://search.maven.org/classic/remotecontent?filepath=com/yahoo/vespa/vespa-http-client/7.391.28/vespa-http-client-7.391.28-jar-with-dependencies.jar
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:41]:
```suggestion
$ curl -L -o vespa-http-client-jar-with-dependencies.jar \
  https://search.maven.org/classic/remotecontent?filepath=com/yahoo/vespa/vespa-http-client/7.391.28/vespa-http-client-7.391.28-jar-with-dependencies.jar
```

File: incremental-search/search-suggestions/README.md [hunk 3]
```diff
@@ -1 +1,65 @@
...
+
+<pre data-test="exec">
+$ git clone --depth 1 https://github.com/vespa-engine/sample-apps.git
+$ cd incremental-search/search-suggestions
+</pre>
+
+**Set up and docker container**
+<pre data-test="exec">
+$ docker pull vespaengine/vespa
+$ docker run -m 6G --detach --name vespa --hostname vespa-example --publish 8080:8080 --publish 19071:19071 vespaengine/vespa
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:18]:
```suggestion
$ docker run -m 6G --detach --name vespa --hostname vespa-example \
   --publish 8080:8080 --publish 19071:19071 \
   vespaengine/vespa
```

File: incremental-search/search-suggestions/README.md [hunk 4]
```diff
@@ -1 +1,65 @@
...
+
+Check respons with:
+<pre data-test="exec">
+$ curl -s --head http://localhost:19071/ApplicationStatus
+</pre>
+
+**Build and deploy the application**
+<pre data-test="exec">
+$ mvn clean package
+$ curl --header Content-Type:application/zip --data-binary @target/application.zip localhost:19071/application/v2/tenant/default/prepareandactivate
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:31]:
```suggestion
$ curl --header Content-Type:application/zip --data-binary @target/application.zip \
  localhost:19071/application/v2/tenant/default/prepareandactivate
```

File: incremental-search/search-suggestions/README.md [hunk 5]
```diff
@@ -1 +1,65 @@
...
+
+Open http://localhost:8080/site/ in a browser.
+
+## General
+
+ ### Streaming search
+
+ [Streaming search](https://docs.vespa.ai/documentation/streaming-search.html) matches documents to queries by streaming through them, similar to grep and does not create precomputed indices. With streaming search only raw data of the documents are stored and and term stemming is not supported.
+
+ Streaming search searches through a documnet collection with a given groupname, so it is important to set a groupname in the id of the documents you intend to use with streaming search.
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:55]:
```suggestion
 Streaming search searches through a document collection with a given groupname, so it is important to set a groupname in the id of the documents you intend to use with streaming search.
```

File: incremental-search/search-suggestions/README.md [hunk 6]
```diff
@@ -1 +1,65 @@
...
+
+Streaming search is slower than indexed search, therefore streaming search is most suitable in situations where the document set is relatively small and the documents themself are short.
+
+### The sample application
+
+In this sample application streaming search is used to implement search suggestions based on users' previous queries. By storing user-input as documents you can get some queries that are not suitable for suggestion. A way to combat this is by filtering out user queries that contain terms that are added to a block list. In this sample application a [document processor](https://docs.vespa.ai/en/document-processing.html) is used to filter out sutch qeries under document feeding.
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:61]:
```suggestion
In this sample application streaming search is used to implement search suggestions based on users' previous queries. By storing user-input as documents you can get some queries that are not suitable for suggestion. A way to combat this is by filtering out user queries that contain terms that are added to a block list. In this sample application a [document processor](https://docs.vespa.ai/en/document-processing.html) is used to filter out such qeries under document feeding. 
```

File: incremental-search/search-suggestions/README.md [hunk 7]
```diff
@@ -1 +1,65 @@
...
+$ git clone --depth 1 https://github.com/vespa-engine/sample-apps.git
+$ cd sample-apps/incremental-search/search-suggestions
+</pre>
+
+**Set up and docker container**
+<pre data-test="exec">
+$ docker pull vespaengine/vespa
+$ docker run -m 6G --detach --name vespa --hostname vespa-example \
+   --publish 8080:8080 --publish 19071:19071 \
+   vespaengine/vespa
```
Comment (chunnoo) [inline incremental-search/search-suggestions/README.md:20]:
```suggestion
$ docker run -m 6G --detach --name vespa --hostname vespa-example \
  --publish 8080:8080 --publish 19071:19071 \
  vespaengine/vespa
```

**Review by chunnoo — APPROVED** [review]

**Review by kkraune — APPROVED** [review]
Great! merging for now, will have another round of fine tuning later

**Pull request merged**